doc/docs/environments.md
---
id: environments
title: Environnements de tests
sidebar_label: Environnements de tests
---

Afin de permettre la réalisation de tests d'intégration, nous avons mis en place un environnement "bac à sable", miroir de la production mais avec une base de données séparée. Vous pouvez créer, modifier ou supprimer autant de BSD's de test que vous voulez sur cet environnement.

Accéder à l'environnement bac à sable:

* [https://api.sandbox.trackdechets.beta.gouv.fr](https://api.sandbox.trackdechets.beta.gouv.fr) (API)
* [https://sandbox.trackdechets.beta.gouv.fr](https://sandbox.trackdechets.beta.gouv.fr) (Interface graphique)


Enfin, il y a également un environnement de "recette" permettant aux équipes Trackdéchets de tester de nouvelles fonctionnalités. Nous n'apportons aucune garantie de stabilité, de disponibilité ou de persistance des données sur cette environnement. Il peut toutefois être utilisé pour tester en avant première des fonctionnalités qui sont annoncées sur le [forum](https://forum.trackdechets.beta.gouv.fr) ou dans la newsletter tech


Accéder à l'environnement de recette:

* [https://api.trackdechets.fr](https://api.trackdechets.fr) (API)
* [https://recette.trackdechets.fr](https://tecette.trackdechets.fr) (interface graphique
)


Changelog.md
# Changelog

Les changements importants de Trackdéchets sont documentés dans ce fichier.

Le format est basé sur [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
et le projet suit un schéma de versionning inspiré de [Calendar Versioning](https://calver.org/).

# Unreleased

#### :rocket: Nouvelles fonctionnalités

- Ajout d'un nouveau champ `packagingInfos` qui viendra remplacer `packagings`, `numberOfPackages` et `otherPackaging`. Ces champs sont encore supportés pour quelques temps mais marqué comme dépréciés. Nous vous invitons à migrer aussi vite que possible. [PR 600](https://github.com/MTES-MCT/trackdechets/pull/600)

#### :boom: Breaking changes

- Suppression de la possibilité pour l'émetteur du bordereau de valider un enlèvement sans signature transporteur dans l'interface Trackdéchets. Cette fonctionnalité utilisait les mutations `markAsSent` et `markAsResent` de l'API qui sont dépréciées. [PR 704](https://github.com/MTES-MCT/trackdechets/pull/704)

#### :bug: Corrections de bugs

- Correction de la mutation `duplicateForm` pour dupliquer l'entreposage provisoire, [PR 700](https://github.com/)
- Correction d'un bug affichant une erreur serveur à la place d'une erreur de validation graphQL lorsque le typage des variables graphQL est erronée [PR 711](https://github.com/MTES-MCT/trackdechets/pull/711)

#### :nail_care: Améliorations

- Amélioration des suggestions d'entreprise lors de la création d'un BSD depuis l'interface, [PR 673](https://github.com/MTES-MCT/trackdechets/pull/673)

#### :memo: Documentation

#### :house: Interne

- Migration du service td-etl dans un projet Github à part [PR 683](https://github.com/MTES-MCT/trackdechets/pull/683)
- Suppression des actions permettant de valider un enlèvement sans signature transporteur dans l'interface Trackdéchets [PR 704](https://github.com/MTES-MCT/trackdechets/pull/704)


# [2020.10.1] 03/11/2020

**Breaking changes**

- Le champ `Form.ecoOrganisme` n'est plus du type `EcoOrganisme` mais du nouveau type `FormEcoOrganisme`.
  Concrètement, le nouveau type ne contient plus les champs `id` et `address`.
  Vous n'êtes pas affecté si vous ne requêtiez pas ces champs ou l'objet `ecoOrganisme`.
- Le type `EcoOrganismeInput` a évolué suite aux changements du champ `Form.ecoOrganisme` :
  ```diff
    input EcoOrganismeInput {
  -   id: ID!
  +   name: String!
  +   siret: String!
    }
  ```
  Vous n'êtes pas affecté si vous ne renseigniez pas l'éco-organisme via les mutations `createForm` ou `updateForm`.

**Changes**

- Refonte de l'interface utilisateur. [PR 469](https://github.com/MTES-MCT/trackdechets/pull/469)
- Ajout du champ `customInfo` à `TransporterInput`, ce qui permet de renseigner cette information via les mutations `createForm`, `updateForm`, `markAsResent`, `markAsResealed`, [PR 417](https://github.com/MTES-MCT/trackdechets/pull/417)
- Suppression du service metabase suite au basculement vers une instance metabase dédiée [PR 453](https://github.com/MTES-MCT/trackdechets/pull/453)
- Ajout du profil d'entreprise "éco-organisme". Ce type d'entreprise peut renseigner ses agréments et signer un BSD à la place du détenteur lorsqu'il est responsable des déchets. [PR 400](https://github.com/MTES-MCT/trackdechets/pull/400)
- Dépréciation des arguments `first` et `skip` sur la query `forms`. A la place, pour paginer utiliser `cursorAfter` et `first` ou `cursorBefore` et `last`. Côté filtres, ajout des arguments `updatedAfter` et `sentAfter` sur la query `forms` pour filtrer par date, `wasteCode` pour filtrer par code déchet, et de `siretPresentOnForm` pour sélectionner des bordereaux ou le SIRET passé apparait [PR 455](https://github.com/MTES-MCT/trackdechets/pull/455)
- Ajout d'un mécanisme de demande de rattachement à un établissement [PR 418](https://github.com/MTES-MCT/trackdechets/pull/418)
- Mise à jour des liens Géorisques cassés [PR 645](https://github.com/MTES-MCT/trackdechets/pull/645)
- Correction d'un bug empêchant l'affichage du dashboard lorsqu'un BSD n'avait pas d'émetteur [PR 644](https://github.com/MTES-MCT/trackdechets/pull/644)
- Correction d'un bug affichant une invitation en attente même quand celle-ci a déjà été acceptée [PR 671](https://github.com/MTES-MCT/trackdechets/pull/671)
- Correction du lien présent dans l'email d'invitation suite à l'action "Renvoyer l'invitation" [PR 648](https://github.com/MTES-MCT/trackdechets/pull/648)
- Champs requis dans le formulaire d'inscription suite à un lien d'invitation [PR 670](https://github.com/MTES-MCT/trackdechets/pull/670)
- Affichage des bordereaux au statut `GROUPED` dans l'onglet "Suivi" du dashboard et corrections de la mutation `markAsSent` sur un BSD de regroupement [PR 672](https://github.com/MTES-MCT/trackdechets/pull/672)
- Correction d'un bug permettant de sceller des bordereaux avec des informations sur le détail du déchet (cadre 3,4,5,6) erronnées ce qui causait des erreurs de validation ultérieures [PR 681](https://github.com/MTES-MCT/trackdechets/pull/681)
- Correction d'un bug empêchant la complétion du BSD suite depuis l'interface [PR 662](https://github.com/MTES-MCT/trackdechets/pull/662)
- Correction d'un bug lors de l'appel à la mutation `markAsTempStored` sans passer le paramètre optionnel `signedAt` [PR 602](https://github.com/MTES-MCT/trackdechets/pull/602)

# [2020.10.1] 05/10/2020

- Ajout d'une limitation de 1000 requêtes possible par une même adresse IP dans une fenêtre de 1 minute, [PR 407](https://github.com/MTES-MCT/trackdechets/pull/407)
- Amélioration de la machine à état permettant de calculer les états possibles du BSD [PR 411](https://github.com/MTES-MCT/trackdechets/pull/411)
- Ajout de la possibilité de pouvoir importer un BSD papier signé, [PR 404](https://github.com/MTES-MCT/trackdechets/pull/404)
- Préservation de la sélection d'entreprise après la création d'un BSD, [PR 410](https://github.com/MTES-MCT/trackdechets/pull/410)

# [2020.09.1] 28/09/2020


pdf/src/settings.js
// Standard page height in pixels
const pageHeight = 842;

// Main form field settings
// coordinates: top/left, mandatory
// fontsize: optional
// maxLength: how many characters do we display, optional
// lineBreakAt: at which position do we insert a linebreak (optional)
// rightAlign: right align the content
const mainFormFieldSettings = {
  currentPageNumber: { x: 530, y: 70, fontSize: 10 },
  totalPagesNumber: { x: 550, y: 70, fontSize: 10 },

  emitterTypeProducer: { x: 53, y: 126, fontSize: 12 },
  emitterTypeAppendix1: { x: 157.5, y: 125.5, fontSize: 12 },
  emitterTypeAppendix2: { x: 51, y: 167, fontSize: 12 },
  emitterTypeOther: { x: 213.5, y: 167, fontSize: 12 },
  readableId: { x: 115, y: 104, fontSize: 10 },
  customId: { x: 300, y: 104, fontSize: 10 },

  ecoOrganismeName: { x: 180, y: 180 },

  emitterCompanySiret: { x: 88, y: 210 },
  emitterCompanyName: { x: 70, y: 219, maxLength: 50 },
  emitterCompanyAddress: { x: 78, y: 230, lineBreakAt: 55, maxLength: 110 },
  emitterCompanyPhone: { x: 65, y: 250 },
  emitterCompanyMail: { x: 64, y: 261 },
  emitterCompanyContact: { x: 125, y: 271 },

  recipientCompanySiret: { x: 342, y: 178 },
  recipientCompanyName: { x: 325, y: 189, maxLength: 50 },
  recipientCompanyAddress: { x: 331, y: 198, lineBreakAt: 55, maxLength: 110 },
  recipientCompanyPhone: { x: 317, y: 219 },
  recipientCompanyMail: { x: 318, y: 229.5 },
  recipientCompanyContact: { x: 378, y: 240.5 },
  recipientCap: { x: 402, y: 261 },
  recipientProcessingOperation: { x: 507, y: 271 },
  temporaryStorageYes: { x: 305, y: 147, fontSize: 12 },
  temporaryStorageNo: { x: 305, y: 159, fontSize: 12 },

  wasteDetailsCode: { x: 155, y: 292, fontSize: 10 },
  wasteDetailsConsistenceSolid: { x: 368, y: 293, fontSize: 12 },
  wasteDetailsConsistenceLiquid: { x: 422, y: 293, fontSize: 12 },
  wasteDetailsConsistenceGaseous: { x: 481, y: 293, fontSize: 12 },
  wasteDetailsName: { x: 172, y: 313 },

  wasteDetailsOnuCode: { x: 45, y: 335 },

  wasteDetailsPackagingsBenne: { x: 132, y: 358, fontSize: 12 },
  wasteDetailsPackagingsCiterne: { x: 179, y: 358, fontSize: 12 },
  wasteDetailsPackagingsGrv: { x: 230, y: 358, fontSize: 12 },
  wasteDetailsPackagingsFut: { x: 272, y: 358, fontSize: 12 },
  wasteDetailsPackagingsOther: { x: 324, y: 358, fontSize: 12 },
  wasteDetailsNumberOfPackages: { x: 505, y: 355 },

  wasteDetailsQuantityReal: { x: 112, y: 380, fontSize: 12 },
  wasteDetailsQuantityEstimated: { x: 170, y: 380, fontSize: 12 },
  wasteDetailsQuantity: { x: 248, y: 377, rightAlign: true },

  traderCompanySiret: { x: 87, y: 410 },
  traderCompanyName: { x: 70, y: 420, maxLength: 50 },
  traderCompanyAddress: { x: 78, y: 430, lineBreakAt: 55, maxLength: 110 },
  traderCompanyPhone: { x: 318, y: 428 },
  traderCompanyMail: { x: 319, y: 438.5 },
  traderCompanyContact: { x: 377, y: 418 },
  traderReceipt: { x: 350, y: 398 },
  traderValidityLimit: { x: 366, y: 408 },
  traderDepartment: { x: 468, y: 398 },

  transporterCompanySiren: { x: 87, y: 474 },
  transporterCompanyName: { x: 70, y: 484, maxLength: 50 },
  transporterCompanyAddress: { x: 78, y: 494, lineBreakAt: 55, maxLength: 110 },
  transporterCompanyPhone: { x: 64, y: 515 },
  transporterCompanyMail: { x: 65, y: 525 },
  transporterCompanyContact: { x: 122, y: 536 },
  transporterReceipt: { x: 350, y: 462 },
  transporterValidityLimit: { x: 366, y: 472 },
  transporterDepartment: { x: 472, y: 462 },
  transporterSentAt: { x: 394, y: 492 },

  senderSentBy: { x: 70, y: 588 },
  senderSentAt: { x: 216, y: 588 },

  recipientCompanySiret10: { x: 86, y: 620 },
  recipientCompanyName10: { x: 72, y: 632, maxLength: 50 },
  recipientCompanyAddress10: { x: 78, y: 642, lineBreakAt: 55, maxLength: 110 },
  recipientCompanyContact10: { x: 124, y: 662 },
  recipientPhoneContact10: { x: 124, y: 662 },
  quantityReceived: { x: 198, y: 673, rightAlign: true },
  receivedBy10: { x: 85, y: 735 },
  receivedAt: { x: 122, y: 683 },
  wasteRefusalReason: { x: 100, y: 704, lineBreakAt: 50 },

  signedAt: { x: 66, y: 746 },

  wasteAccepted: { x: 116, y: 696, fontSize: 12 },
  wasteNotAccepted: { x: 161, y: 696, fontSize: 12 },

  processingOperationDone: { x: 338, y: 620 },
  processingOperationDescription: { x: 342, y: 640 },
  processedBy: { x: 322, y: 680 },
  processedAt: { x: 318, y: 692 },

  nextDestinationProcessingOperation: { x: 154, y: 776.5 },
  nextDestinationCompanySiret: { x: 88, y: 788 },
  nextDestinationCompanyName: { x: 69, y: 798, maxLength: 50 },
  nextDestinationCompanyAddress: {
    x: 79,
    y: 808,
    lineBreakAt: 55,
    maxLength: 110
  },
  nextDestinationCompanyContact: { x: 380, y: 787 },
  nextDestinationCompanyPhone: { x: 318, y: 798 },
  nextDestinationCompanyMail: { x: 318, y: 808 },
  isMultimodal: { x: 303, y: 536, fontSize: 12 }
};

// coordinates of each stamp image
const imageLocations = {
  transporterSignature: { x: 450, y: 525 },
  emitterSignature: { x: 450, y: 590 },
  processingSignature: { x: 450, y: 732 },
  receivedSignature: { x: 210, y: 742 },
  exemptionStamp: { x: 400, y: 520 },
  noTraceabilityStamp: { x: 300, y: 740 },
  watermark: { x: 0, y: 800 },
  tempStorerReceptionSignature: { x: 195, y: 310 },
  tempStorerSentSignature: { x: 420, y: 590 },
  tempStorageTransporterSignature: { x: 420, y: 480 },
  takenOverSignature: { x: 420, y: 695 }
};

// Temporary Storage detail form field settings
const temporaryStorageDetailsFieldSettings = {
  currentPageNumber: { x: 530, y: 70, fontSize: 10 },
  totalPagesNumber: { x: 550, y: 70, fontSize: 10 },
  formReadableId: { x: 220, y: 95, fontSize: 10 },

  tempStorerQuantityReal: { x: 125, y: 190, fontSize: 12 },
  tempStorerQuantityEstimated: { x: 190, y: 190, fontSize: 12 },
  tempStorerQuantityReceived: { x: 222, y: 193 },

  wasteAccepted: { x: 116, y: 223, fontSize: 12 },
  wasteNotAccepted: { x: 161, y: 223, fontSize: 12 },

  tempStorerWasteRefusalReason: { x: 100, y: 240, lineBreakAt: 50 },
  tempStorerReceivedAt: { x: 125, y: 213 },
  tempStorerSignedAt: { x: 75, y: 273 },
  tempStorerCompanySiret: { x: 90, y: 151 },
  tempStorerCompanyName: { x: 75, y: 161, maxLength: 50 },
  tempStorerCompanyAddress: { x: 80, y: 170, lineBreakAt: 55, maxLength: 110 },

  destinationIsFilledByEmitter: { x: 293, y: 284, fontSize: 12 },
  destinationCompanySiret: { x: 335, y: 140 },
  destinationCompanyName: { x: 320, y: 150, maxLength: 50 },
  destinationCompanyAddress: {
    x: 325,
    y: 161,
    lineBreakAt: 55,
    maxLength: 110
  },
  destinationCompanyPhone: { x: 313, y: 180 },
  destinationCompanyMail: { x: 313, y: 190 },
  destinationCompanyContact: { x: 372, y: 202 },
  destinationCap: { x: 400, y: 222 },
  destinationProcessingOperation: { x: 505, y: 232 },

  wasteDetailsOnuCode: { x: 50, y: 340 },
  wasteDetailsPackagingsBenne: { x: 132, y: 370, fontSize: 12 },
  wasteDetailsPackagingsCiterne: { x: 179, y: 370, fontSize: 12 },
  wasteDetailsPackagingsGrv: { x: 230, y: 370, fontSize: 12 },
  wasteDetailsPackagingsFut: { x: 279, y: 370, fontSize: 12 },
  wasteDetailsPackagingsOther: { x: 319, y: 370, fontSize: 12 },
  wasteDetailsNumberOfPackages: { x: 520, y: 369 },

  wasteDetailsQuantityReal: { x: 112, y: 390, fontSize: 12 },
  wasteDetailsQuantityEstimated: { x: 170, y: 390, fontSize: 12 },
  wasteDetailsQuantity: { x: 248, y: 387, rightAlign: true },

  transporterCompanySiret: { x: 90, y: 430 },
  transporterCompanyName: { x: 75, y: 440, maxLength: 50 },
  transporterCompanyAddress: { x: 80, y: 450, lineBreakAt: 55, maxLength: 110 },
  transporterCompanyPhone: { x: 65, y: 470 },
  transporterCompanyMail: { x: 65, y: 490 },
  transporterCompanyContact: { x: 125, y: 501 },
  transporterReceipt: { x: 350, y: 417 },
  transporterDepartment: { x: 472, y: 417 },
  transporterValidityLimit: { x: 366, y: 425 },
  transporterSentAt: { x: 388, y: 455 },

  tempStoredFormSignedBy: { x: 75, y: 562 },
  tempStoredFormSignedAt: { x: 215, y: 562 }
};

// appendix2 header field settings
const appendixHeaderFieldSettings = {
  readableId: { x: 240, y: 160, fontSize: 10 },
  emitterCompanySiret: { x: 103, y: 186, fontSize: 10 },
  emitterCompanyName: { x: 84, y: 197, fontSize: 10 },
  emitterCompanyAddress: { x: 92, y: 208, fontSize: 10, lineBreakAt: 40 },
  emitterCompanyMailName: { x: 372, y: 186, fontSize: 10 },
  emitterCompanyPhone: { x: 306, y: 197, fontSize: 10 },
  emitterCompanyMail: { x: 305, y: 208, fontSize: 10 },
  emitterCompanyContact: { x: 375, y: 186, fontSize: 10 }
};

// appendix2 attached form field settings
const appendixFieldSettings = {
  numbering: { x: 142, y: 243, fontSize: 10 },
  initialEmitter: { x: 142, y: 243, fontSize: 10 },
  readableId: { x: 392, y: 244, fontSize: 10 },
  emitterCompanySiret: { x: 104, y: 256, fontSize: 10 },
  emitterCompanyName: { x: 84, y: 267, fontSize: 10, maxLength: 29 },
  emitterCompanyAddress: { x: 94, y: 278, fontSize: 10, lineBreakAt: 38 },
  emitterCompanyPhone: { x: 76, y: 302, fontSize: 10 },

  emitterCompanyMail: { x: 76, y: 313, fontSize: 10 },
  emitterCompanyContact: { x: 143, y: 326, fontSize: 10 },
  wasteDetailsCode: { x: 360, y: 267, fontSize: 10 },
  wasteDetailsName: { x: 430, y: 279, fontSize: 10 },
  wasteDetailsQuantityReal: { x: 327, y: 302, fontSize: 10 },
  quantityReceived: { x: 468, y: 302, fontSize: 10, rightAlign: true },
  receivedAt: { x: 350, y: 326, fontSize: 10 }
};

// vertical offset to be applied to each appending sub form coordinates
const appendixYOffsets = [0, 104, 208, 313, 418];

const transportSegmentSettings = {
  segmentNumber: { x: 162, y: 628, fontSize: 10 },
  transporterCompanySiren: { x: 90, y: 640, fontSize: 10 },
  transporterCompanyName: { x: 71, y: 650, fontSize: 10, maxLength: 50 },
  transporterCompanyAddress: {
    x: 77,
    y: 660,
    fontSize: 10,
    lineBreakAt: 55,
    maxLength: 110
  },
  transporterCompanyPhone: { x: 62, y: 680, fontSize: 10 },
  transporterCompanyMail: { x: 63, y: 691, fontSize: 10 },
  transporterCompanyContact: { x: 122, y: 702, fontSize: 10 },
  transporterReceipt: { x: 350, y: 628, fontSize: 10 },
  transporterDepartment: { x: 468, y: 628, fontSize: 10 },
  transporterValidityLimit: { x: 364, y: 638, fontSize: 10 },
  mode: { x: 368, y: 650, fontSize: 10 },
  takenOverAt: { x: 391, y: 659, fontSize: 10 },
  takenOverBy: { x: 332, y: 678, fontSize: 10 }
};

exports.pageHeight = pageHeight;
exports.mainFormFieldSettings = mainFormFieldSettings;
exports.temporaryStorageDetailsFieldSettings = temporaryStorageDetailsFieldSettings;
exports.imageLocations = imageLocations;
exports.appendixHeaderFieldSettings = appendixHeaderFieldSettings;
exports.appendixFieldSettings = appendixFieldSettings;
exports.appendixYOffsets = appendixYOffsets;
exports.transportSegmentSettings = transportSegmentSettings;
